tests/test_api.py

```diff
@@ -2520,6 +2527,10 @@ def _check_attachment(self, data, attachment_id, additional_fields):
                                            [["id", "is", attachment_id]],
                                            list(expected_fields))
 
+        # remove images before comparison
+        if "this_file" in attachment_data:
+            attachment_data["this_file"].pop("url")
+            data["this_file"].pop("url")
```

Comment:
Is it probable that "this_file" key doesn't exist in "data" dict? so we should check if "this_file" exists in "data"